gh-135788: Support NETRC environment variable in `netrc` (#135802)
python/cpython: Doc/library/netrc.rst

.. class:: netrc([file])

   A :class:`~netrc.netrc` instance or subclass instance encapsulates data from  a netrc
   file. The initialization argument, if present, specifies the file to parse. If no
   argument is given, it will look for the file path in the :envvar:`!NETRC` environment variable.
   If that is not set, it defaults to reading the file :file:`.netrc` in the user's home

>    directory -- as determined by :func:`os.path.expanduser`. If the file cannot be found,

Reviewer: Please add a `versionadded:: next` entry below indicating that we now support the NETRC environment variable. Also, add a NEWS entry and update Doc/whatsnew/3.15.rst. TiA.

Author: Done. I’ve added the `.. versionadded:: next` directive, but I’m not completely sure if I’m using it as intended. Please let me know if any changes are needed.

>    argument is given, it will look for the file path in the :envvar:`!NETRC` environment variable.
>    If that is not set, it defaults to reading the file :file:`.netrc` in the user's home
>    directory -- as determined by :func:`os.path.expanduser`. If the file cannot be found,

Reviewer: ```suggestion
   argument is given, it will look for the file path in the :envvar:`!NETRC` environment variable,
   before defaulting to reading the file :file:`.netrc` in the user's home
   directory -- as determined by :func:`os.path.expanduser`. If the file cannot be found,
```

      can contain arbitrary characters, like whitespace and non-ASCII characters.
      If the login name is anonymous, it won't trigger the security check.

   .. versionadded:: next

>       :class:`netrc` try to use the value of the :envvar:`NETRC` environment variable
>       if when *file* is not passed as argument, before falling back to the user's
>       :file:`.netrc` file in the home directory.

Reviewer: ```suggestion
      Added support for the :envvar:`!NETRC` environment variable.
```
